api/api.go

```diff
@@ -77,33 +77,44 @@ func (api *API) enablePublicEndpoints() {
 }
 
 func (api *API) enablePrivateEndpoints() {
-	r := chi.NewRouter()
+	// This is a hack to work around the issue whereby a chi.Router cannot set middleware after a route has been added
```

Comment:
I'm not sure why this is necessary, if it's to enable something you've done in the component tests I'd rather not introduce hacks into our main service code to enable that.. It would be better if we could find a way so this wasn't necessary.

Reply:
This one is a bit of a strange one. It's not directly related to component tests - as in the component tests will still work without the change I made - but I was looking at it when doing the component tests, since a component test can dynamically use the 'private endpoints are enabled/not enabled' (the reason the component tests will work is that the component is reset and re-initialised after every scenario, so the router will be re-initialised as well). But I made the change in an attempt to remove any prior knowledge of the state of the router when adding the private routes. Probably easier to have a conversation on this one, but I can remove it - I wanted to get some feedback on it first, and discuss what was there beforehand